app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx

```diff
@@ -110,6 +110,7 @@ function App(props: AppProps) {
 function AppRouter(props: AppProps) {
   const { logger, onAuthenticated } = props;
   const navigate = router.useNavigate();
+  const mainPageUrl = new URL(window.location.href)
```

Comment:
does this offer any advantages over using `window.location.href`?

Reply:
window.location.href returns string and we want mainPageUrl to be URL typed

Reply:
yeah, but is there a reason `mainPageUrl` is supposed to be a `URL` object rather than a string?